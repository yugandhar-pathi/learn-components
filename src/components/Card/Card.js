import PropTypes from 'prop-types'

export const Card = ({heading, content, picture, buttonName, altText}) => {
    const cardStyle = {
        boxShadow: "0 4px 8px 0 rgba(0,0,0,0.2)",
        transition: "0.3s",
        width: "20%",
        margin: "1%"
    }

    const divStyle = {
        display: "flex",
        flexDirection: "column",
        padding: "6%"
    }


    return(
       <div style={cardStyle}>
           <div style={divStyle}>
               <h2><b>{heading}</b></h2>
               {
                (content && picture) ? 
                <div>
                    <div>{content}</div>
                    <img src={picture} alt={altText} width="100%"/>
                </div> :
                <div>{content || <img src={picture} alt={altText} width="100%"/>}</div>
               }
               <button style={{border:"none", color:"blue"}}>{buttonName}</button>  
            </div>         
        </div>
    )
}

Card.propTypes = {
    heading: PropTypes.string.isRequired,
    content: PropTypes.string,
    buttonName: PropTypes.string.isRequired,
    altText: PropTypes.string
}