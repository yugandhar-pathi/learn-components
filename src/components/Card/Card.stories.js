import React from "react";
import { Card } from "./Card";

export default{
    component : Card,
}

export const basicCard = () => <Card heading="Basic card" content="this is a basic card" buttonName="Learn More"></Card>
export const pictureCard = () => <Card heading="Basic card" picture="https://www.bing.com/th?id=OIP.Ky0Jd3zUxcwi7S5XWV2bxAHaF7&w=279&h=223&c=8&rs=1&qlt=90&o=6&dpr=1.25&pid=3.1&rm=2" altText="Rose" buttonName="Learn More"></Card>
export const complexCard = () => <Card heading="Basic card" content="this is a basic card" picture="https://www.bing.com/th?id=OIP.Ky0Jd3zUxcwi7S5XWV2bxAHaF7&w=279&h=223&c=8&rs=1&qlt=90&o=6&dpr=1.25&pid=3.1&rm=2" altText="Rose" buttonName="Learn More"></Card>