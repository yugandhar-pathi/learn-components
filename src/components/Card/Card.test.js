import {render, screen} from '@testing-library/react';
import {Card} from './Card';

test('Card component', ()=>{
    render(<Card heading="header" dataTestId="test-card" buttonName="my button"/>)
    expect(screen.getByTestId("test-card").textContent).toBe("headermy button")
})

test('Card snapshot', ()=>{
    render(<Card heading="header" content="my Content" picture="image" buttonName="my Button" altText="alternate text" dataTestId="test-card"/>)
    expect(screen.getByTestId("test-card")).toMatchSnapshot()
})