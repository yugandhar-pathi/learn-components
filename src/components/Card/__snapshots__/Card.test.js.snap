// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshot 1`] = `
<div
  class="sc-bczRLJ dzDZer"
  data-testid="test-card"
>
  <div
    class="sc-gsnTZi hyUOGO"
    data-testid="col-test-card"
  >
    <h2
      data-testid="header-test-card"
    >
      <b>
        header
      </b>
    </h2>
    <div>
      <div>
        my Content
      </div>
      <img
        alt="alternate text"
        src="image"
        width="100%"
      />
    </div>
    <button
      data-testid="button-test-card"
      style="color: blue;"
    >
      my Button
    </button>
  </div>
</div>
`;
